gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionProps.flex

```diff
@@ -23,52 +23,52 @@ import com.intellij.psi.TokenType;
-
-<YYINITIAL> {COLON}                                         { yybegin(WAITING_NAME); return VersionPropsTypes.COLON; }
-
-<WAITING_NAME> {KEY_CHARACTER}+                             { yybegin(WAITING_VERSION); return VersionPropsTypes.NAME_KEY; }
-
-<WAITING_NAME> {WHITE_SPACE}+                               { yybegin(WAITING_NAME); return TokenType.WHITE_SPACE; }
-
-<WAITING_NAME> {CRLF}({CRLF}|{WHITE_SPACE})+                { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
 
-<WAITING_VERSION> {EQUALS}                                  { yybegin(WAITING_VALUE); return VersionPropsTypes.EQUALS; }
+<YYINITIAL> {COMMENT}                            { yybegin(YYINITIAL); return VersionPropsTypes.COMMENT; }
+<YYINITIAL> {GROUP_PART}                         { yybegin(YYINITIAL); return VersionPropsTypes.GROUP_PART; }
+<YYINITIAL> {DOT}                                { yybegin(YYINITIAL); return VersionPropsTypes.DOT; }
+<YYINITIAL> {WHITE_SPACE}*{DOT}{WHITE_SPACE}*    { yybegin(INVALID_VALUE); return TokenType.BAD_CHARACTER; }
+<YYINITIAL> {COLON}                              { yybegin(WAITING_NAME); return VersionPropsTypes.COLON; }
+<YYINITIAL> {WHITE_SPACE}*{COLON}{WHITE_SPACE}*  { yybegin(INVALID_VALUE); return TokenType.BAD_CHARACTER; }
 
-<WAITING_VERSION> {WHITE_SPACE}+                            { yybegin(WAITING_VERSION); return TokenType.WHITE_SPACE; }
+<WAITING_NAME> {KEY_CHARACTER}                   { yybegin(WAITING_VERSION); return VersionPropsTypes.NAME_KEY; }
+<WAITING_NAME> {WHITE_SPACE}+                    { return TokenType.WHITE_SPACE; }
 
-<WAITING_VERSION> {CRLF}({CRLF}|{WHITE_SPACE})+             { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
+<WAITING_VERSION> {EQUALS}                       { yybegin(WAITING_VALUE); return VersionPropsTypes.EQUALS; }
+<WAITING_VERSION> {WHITE_SPACE}+                 { return TokenType.WHITE_SPACE; }
 
-<WAITING_VALUE> {WHITE_SPACE}+                              { yybegin(WAITING_VALUE); return TokenType.WHITE_SPACE; }
+<WAITING_VALUE> {WHITE_SPACE}+                   { return TokenType.WHITE_SPACE; }
+<WAITING_VALUE> {VALUE}                          { yybegin(WAITING_COMMENT); return VersionPropsTypes.VERSION; }
 
-<WAITING_VALUE> {VALUE_CHARACTER}+                          { yybegin(INVALID_VALUE); return VersionPropsTypes.VERSION; }
+<WAITING_COMMENT> {WHITE_SPACE}*{COMMENT}        { yybegin(INVALID_VALUE); return VersionPropsTypes.COMMENT; }
+<WAITING_COMMENT> [^\n]+                         { return TokenType.BAD_CHARACTER; }
 
-<INVALID_VALUE> [^\n]+                                      { return TokenType.BAD_CHARACTER; }
+<INVALID_VALUE> [^\n]+                           { return TokenType.BAD_CHARACTER; }
 
-({CRLF}|{WHITE_SPACE})+                                     { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
+({CRLF}|{WHITE_SPACE})+                          { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
```

Comment:
I think this will allow for the start of the line to be a white space: 
` com.palantir....=1.0.0`

Reply:
good spot! - Ive fixed it now but still allowed white space before comments